main.py

```diff
@@ -154,6 +156,20 @@ def is_valid_tag(tag: Optional[str]) -> bool:
     return content
 
 
+def get_cover_art(filename: str) -> Optional[discord.File]:
+    tags = TinyTag.get(filename, image=True)
+    # get image data as bytes, make sure it doesn't go over 8MB
+    # This is a safe lower bound on the Discord upload limit of 8MiB
+    image_data = tags.get_image()
+    if image_data is None or len(image_data) > 8_000_000:
+        return None
+    image = Image.open(BytesIO(image_data))
+    cover_filename = f"cover.{image.format}".lower()
+    cover_filepath = path.join(attachment_directory_filepath, cover_filename)
+    image.save(cover_filepath)
+    return discord.File(cover_filepath, filename=cover_filename)
```

Comment:
We can actually upload the bytes as a file without needing to save it to disk first, by reusing our `BytesIO` object:

```python
def get_cover_art(filename: str) -> Optional[discord.File]:
    # Get image data as bytes
    tags = TinyTag.get(filename, image=True)
    image_data = tags.get_image()

    # Make sure it doesn't go over 8MB.
    # This is a safe lower bound on the Discord upload limit of 8MiB
    if image_data is None or len(image_data) > 8_000_000:
        return None

    # Get a file pointer to the bytes
    image_bytes_fp = BytesIO(image_data)

    # Read the filetype of the bytes and discern the appropriate file extension
    image = Image.open(image_bytes_fp)

    # Wind back the file pointer in order to read it a second time
    image_bytes_fp.seek(0)

    # Make up a filename
    cover_filename = f"cover.{image.format}".lower()

    # Create a new discord file from the file pointer and name
    return discord.File(image_bytes_fp, filename=cover_filename)
```